### PR TITLE
public grpc method descriptors in generated code #1273

### PR DESCRIPTION
**Pull request by justcoon**
grpc MethodDescriptors as public val-s in generated code

References #1237

### CODE REVIEW

**Comment by lightbend-cla-validator** [comment]
At least one pull request committer is not linked to a user. See https://help.github.com/en/articles/why-are-my-commits-linked-to-the-wrong-user#commits-are-not-linked-to-any-user

**Comment by raboof** [comment]
OK TO TEST

File: codegen/src/main/twirl/templates/ScalaClient/Client.scala.txt
Comment (raboof) [inline codegen/src/main/twirl/templates/ScalaClient/Client.scala.txt:105]: It looks like removing this object caused MiMa to flag a potential binary incompatibility. I think this might probably be all-right since this object is private anyway, but perhaps double-check this apply method would not be used from user code. If so then we can add a mima exclusion rule to specify this is all-right.

Comment (justcoon) [reply codegen/src/main/twirl/templates/ScalaClient/Client.scala.txt:105]: i removed it, because it was failing on compilation, 
after i moved method descriptors, there was just apply function which was not used

**Comment by ignasi35** [comment]
LGTM once the MiMa issue is fixed.

**Comment by justcoon** [comment]
@raboof: do you have some idea/proposal, how to solve MiMa issue?

**Comment by ignasi35** [comment]
> @raboof: do you have some idea/proposal, how to solve MiMa issue?

Hi @justcoon, you have several options:

1. restore the `apply()` method
2. filter that method from the mima verification using the `ProblemFilter`detailed on the [error message](https://travis-ci.com/github/akka/akka-grpc/jobs/478730662#L328-L333) following the example of [mima filters in other modules](https://github.com/akka/akka-grpc/blob/master/runtime/src/main/mima-filters/1.0.x.backwards.excludes/akka-http2-client.excludes)

But I think @raboof's recommendation to first validate the `apply()` method is not used from user code would be best.

**Review by raboof — APPROVED** [review]
Thanks!

**Comment by raboof** [comment]
this means a binary incompatibility between the client and the api trait, but I think it's reasonable to require those to be generated with the same version of Akka gRPC.

**Pull request merged**
